### ISSUE TITLE
iOS Version Of GC4iOS

### ISSUE THREAD

**Issue by mk8itra1n** [WilliamLCobb/GC4iOS#14]
@WilliamLCobb The app works great on my iPhone 6 Plus but there are a few issues I see. The Launch With JIT (jailbroken only) option does not work and crashes my device back to the home screen but I am actually jailbroken. Also I am not able to control any games at all when they boot as they don't respond but it's just an alpha so I didn't expect it to. The games do not fit the resolution of my device correctly as well and the graphics flicker. There is no inapp downloader or any type of customizable settings.Just wanna know when you gonna be tackling all these issue. This is amazing work for an alpha btw!!! Didn't even expect to make it to the title screen so this emulator is definitely off to a good start👍 
![image](https://cloud.githubusercontent.com/assets/12947869/15988294/a000bede-3018-11e6-99de-dda20ae7e0ef.png)

**Comment by nfsvsgta** [comment]
The resolution seems to be high. The aspect ratio may not be correct

**Comment by mk8itra1n** [comment]
Here's some things that should be doable for the iOS version of GC4iOS as of the Alpha Version...
-Improved User Interface
-Getting the games to respond to the onscreen touch controls
-Sound Support 
-Configurable Options just like iNDS
-In App Rom Downloading Browser
-Turbo Boosting with Interpreter Engine for FPS boost just like iNDS
-JIT for non jailbroken/jailbroken devices both 32 bit and 64 bit
-Dropbox Sync
-OpenGL Support
-Fix Aspect Ratio and Resolution for each respective iDevice
-Boxart for each game currently added to the emulator
@WilliamLCobb any ETA for any of these things? Knowing you your probably hammering away at some of these things on the list lmao👍

**Comment by nfsvsgta** [comment]
And the support for wii

**Comment by nfsvsgta** [comment]
@williamlcobb could u also do some contributions to Play! Emulator? The ps2 emulator since ur good in jit and stuff and I believe u can bring massive improvements to it.

**Comment by jakeb072001** [comment]
Wii games already work (for some games at least)
But it is very, very slow

**Comment by MaquinaDelMisterio** [comment]
We really need people to control creating issues posts. Everyday they create a new one with the same info. It would be better if we use the exting ones.

**Comment by mk8itra1n** [comment]
Most of the Issues on here are literally the same so I see what you mean. My issue I created here is to keep updates for everyone and myself about development of the emulator as well as reporting any bugs I see in the app as I use it and improvements we could realistically see to GC4iOS

**Comment by MaquinaDelMisterio** [comment]
@mk8itra1n  I liked that you clarified it. And know that you say that, maybe creating like sections it would be better just to keep things arrenged.

**Comment by MaquinaDelMisterio** [comment]
@WilliamLCobb Saw you replied to something recently and i just wanted to ask you, why only when connected to xcode, games dont stay on blackscreen on non jailbroken devices? Theres a way i can run them, without Xcode?

**Comment by mk8itra1n** [comment]
@WilliamLCobb And apparently there is a fix for JIT on jail broken devices but it requires Xcode and a specific code line edit to fix. Just wanna know when that JIT fix gonna be pushed to Cydia

**Comment by newhacker1746** [comment]
Mario galaxy working, 9.1 pangu 6+
1 fps, i'll test other games too soon. Jit does in fact crash.
![image](https://cloud.githubusercontent.com/assets/19787725/15994028/78997c42-30c6-11e6-8fef-990b7fc7e616.png)

**Comment by MaquinaDelMisterio** [comment]
@mk8itra1n You are right, there was a line to run jit, but it needs to be fixed anyway. Maybe soon he will fix it.

**Comment by agentgul** [comment]
I just want to swoop in and say that Will has tried very hard for a long time to make jit work on jailbroken devices, and because the ios 9 jailbreak isn't "authentic", don't expect it to happen.

**Comment by MaquinaDelMisterio** [comment]
Maybe he can do what dev of ppsspp did, an irc interpreter core. Im not sure if that will help on gc games, but it works at full speed on ppsspp. Lets see what happens.

**Comment by WilliamLCobb** [comment]
Dolphin's JIT actually works on jailbroken devices now. I just need help from an experienced dev to fix all of the bugs. JIT is also 100% possible on jailed devices and should always be. It will require some specific fixes and will be slower than jailbroken JIT, but it will work once enough time is put in.

**Comment by nfsvsgta** [comment]
@jpd002 and developers of ppsspp ?

**Comment by nfsvsgta** [comment]
@WilliamLCobb And nfs underground stays here ( 10 mins passed and still here).  Do u have any idea what's causing this? 
![image](https://cloud.githubusercontent.com/assets/19260668/16012649/d769e42c-31a3-11e6-8ec0-49415a03c1da.jpg)

**Comment by newhacker1746** [comment]
JIT crashes on launch starfield and mario galaxy. Built latest.
Normal launch still works. But strangely the buttons no longer work to change stars or do anything in starfield.dol

**Comment by mk8itra1n** [comment]
Every game I booted up doesn't respond to the controls. The graphics are emulated perfectly but it does need JIT for a speed boost and maybe some tweaks here and there to make it playable

**Comment by newhacker1746** [comment]
Let's hope William isn't getting overwhelmed by all the work. But it's been a while since anyone replied or anything

**Comment by mk8itra1n** [comment]
@WilliamLCobb if you out there bro don't rush or feel overwhelmed with all of these projects. Take your time and get all the help you need. No need for rushing bro. Good things come to those who are patient

**Comment by nfsvsgta** [comment]
https://github.com/AlexAltea/nucleus
@williamlcobb if you have time can u give a look at this PS3 emulator for iOS? Yes for ios

**Comment by WilliamLCobb** [comment]
Thanks for the encouragement. It is a lot of work and I also work a full time job and a few other apps to pay rent, but it's not too overwhelming. Dolphin on iOS will be merged to the official trunk soon and I'm going to really start working on it then.

As for a timeline, after Dolphin 5.0 is released they will let me merge my iOS version. It was planned to be last weekend but it should happen this weekend or the next. 

This was really just meant for developers at the moment, that's why it's pretty buggy. Once we get JIT to work you guys will be essential for helping me beta test.

**Comment by nfsvsgta** [comment]
Dolphin 5 is out now by the way

**Comment by MaquinaDelMisterio** [comment]
it work for Android?

**Comment by ghost** [comment]
@MaquinaDelMisterio There is already a version of dolphin for android

**Comment by mk8itra1n** [comment]
@WilliamLCobb If your out there bro Dolphin 5 is out and boy is it an amazing release. My games on the PC version run so much better now. So now what's the next step for GC4iOS/DolphiniOS now since Dolphin 5.0 is out?

**Comment by nfsvsgta** [comment]
@williamlcobb I hear the jit has been fixed. Is it true and when will we see an update for gc4ios ? Thanks

**Comment by newhacker1746** [comment]
No merge into dolphin yet, has the project been abandoned?

**Comment by nfsvsgta** [comment]
Kind of since there is another developer who has fixed jit. Don't know whether OpenGL though. Will seems to be helping him since he closed the pull request and the other developer is working on his pull request.

**Comment by WilliamLCobb** [comment]
I'm still working on it. JIT and the OpenGL renderer now done. The emulator runs much faster but it's not fully utilizing the CPU (so the perceived speed up is not very much). Once that is fixed some games should be playable on very fast devices. 

The last thing to fix is fastmem. It's going to be difficult due to the iPhones limited address space but once done should provide a significant speed up. 

Once the CPU scheduling problem has been fixed, I'll publish a new update

Also JIT works on jailed and jailbroken phones.

**Comment by zakusa** [comment]
hi will do you know if mario galaxy wil be able to run fast on ipad air 2 once your done with emulator ya its sad that i got ipad air 2 and now theres ipad pro for the same price as ipad air 2 and i cant even upgrade

**Comment by newhacker1746** [comment]
Im away from home and cannot currently run xcode to test the new version. Ill try the cydia version

**Comment by newhacker1746** [comment]
Not updated yet though.

**Comment by nfsvsgta** [comment]
Updated but I don't have enough space to do any testing yet so just as a foreword what is the directory for placing roms?

**Comment by MaquinaDelMisterio** [comment]
Github isnt updated, or it is?

**Comment by nfsvsgta** [comment]
Tested. Games still crash

**Comment by MaquinaDelMisterio** [comment]
Jailed iOS Device?

**Comment by nfsvsgta** [comment]
Games which ran on previous update crash now :P. It's ios 8.3 jailbroken

**Comment by MaquinaDelMisterio** [comment]
I will report on jailed devices. Hope it doesnt crash.

**Comment by nfsvsgta** [comment]
Thanks for the new update but after installation it won't show up on main screen

**Comment by WilliamLCobb** [comment]
It's the same problem that was causing crashes before. I'll update it tonight to fix it. It should work fine when compiled from Xcode

**Comment by nfsvsgta** [comment]
Well on 0.1.2.1 after installation app won't show up. Hopefully this isn't the update you were talking about of releasing tonight

**Comment by newhacker1746** [comment]
Okay. Could you please try to update the cydia version? As I have no access to xcode (away from home) have just my iphone.

**Comment by MaquinaDelMisterio** [comment]
Im on iPad Pro 12" iOS 9.3.2 and it crashes. I have to be connected to Xcode to run a game which also dont load. Smash Bros stays on white screen, MK DDS shows yellow lines and freezes. Im going to try Luigis Mansion.

**Comment by newhacker1746** [comment]
Thanks. Downloading

**Comment by MaquinaDelMisterio** [comment]
Okey, luigis mansion tis working but its pretty slow on menu and cinematics. I dont reccomend it right now on jailed devices, need xcode to run it without crashing.

**Comment by newhacker1746** [comment]
Cydia version doesnt show up, it did before?

**Comment by zakusa** [comment]
guys will mario galaxy run full speed on ipad air 2 when emulator done

**Comment by newhacker1746** [comment]
The info.plist has several mistakes, I tried to add a cfbundle identifier that did not work. I tried to use open 
"open com.MyApp.MyCompany" but didnt find the id I used "com.test123.test123" I tried switching it with the old version's plist. Didnt work.

**Comment by mk8itra1n** [comment]
After updating just like @nfsvsgta my GC4iOS app no longer shows up even tho the update from Cydia was successful. I'm on a jailbroken iPhone 6 Plus running iOS 9.0.2

**Comment by newhacker1746** [comment]
Ok guys got it just used old info.plist changed version 1.0 to 1.1 and renamed the CFBundleIdentifier to something different, ran uicache in terminal and got it. Also gave executable permissions to the executable inside .app
Crash on running starfield or mario. Maybe just cause I have 9.1 and not 9.0.2 jailbroken?

**Comment by mk8itra1n** [comment]
Unfortunately that method didn't work for me on iOS 9.0.2 jailbroken. Looks like I'll have to wait for that GC4iOS update on Cydia tonight and hopefully the issue will be fixed

**Comment by zakusa** [comment]
does anyone know if mario galaxy will run full speed on ipad air 2 when emulator done

**Comment by Fadi5555** [comment]
Same here after update this emu to latest update it didn't show on home screen anymore?
When @WilliamLCobb gonna fix this issue. Also can you give me the link to download this emu for non jailbroken device please?

**Comment by nfsvsgta** [comment]
It's on ipwnstore

**Comment by Fadi5555** [comment]
Is it free or not?

**Comment by MaquinaDelMisterio** [comment]
Its not, and i wouldnt pay 10 bucks to install that app. Right now its not working very well on non jailbroken devices needs speed fixes. Its unplayable

**Comment by WilliamLCobb** [comment]
Cydia version should be fixed now

**Comment by MaquinaDelMisterio** [comment]
@WilliamLCobb You have a dedicated repo, or its on Bigboss repo? I would like to try the Cydia version.

**Comment by WilliamLCobb** [comment]
www.williamlcobb.com/repo

**Comment by MaquinaDelMisterio** [comment]
Thanks Bud!

**Comment by nfsvsgta** [comment]
Nfs underground stays on black screen. Intro of nfs hp 2 runs on full speed

**Comment by MaquinaDelMisterio** [comment]
I also tried out Cydia version, and also crash when i try to load a rom. I dont get it, why i have to be connected to xcode to run games. By the way, @Fadi5555 i have tried both versions, Cydia and Xcode and both crash on jailed devices, i really dont recommend you to pay 10bucks for this app, yet. I can share with you the deb file if you want to. You just need Instasign to convert it to .ipa

**Comment by DJRuckASS** [comment]
Hey @WilliamLCobb when I go to load a rom it just keeps me on this white page showing the buttons but it doesn't work. I'm on a jailbroken iOS device

**Comment by DJRuckASS** [comment]
I tried running Spyro a Hero's Tail and it just froze me on a black screen showing the buttons

**Comment by nfsvsgta** [comment]
All games which ran on prev version crash now

**Comment by nfsvsgta** [comment]
Or don't run. Games get stuck in menu etc.

**Comment by DJRuckASS** [comment]
Ok

**Comment by nfsvsgta** [comment]
![image](https://cloud.githubusercontent.com/assets/19260668/16543838/7753d224-4106-11e6-8f39-781fd1a173df.jpg)
Nfs underground on black screen

**Comment by nfsvsgta** [comment]
Animal crossing stay on white screen with sound behind.

**Comment by nfsvsgta** [comment]
in the prev update there were visuals but no sound lol

**Comment by nfsvsgta** [comment]
Duke nukem gets stuck on the copyright notice screen whereas in prev update it reached the main menu

**Comment by nfsvsgta** [comment]
Moreover speed seems fine since nfs hp2 intro movie runs at almost full screen. There is nothing but text in its main menu. Game crashes as soon as race starts

**Comment by nfsvsgta** [comment]
I mean full speed

**Comment by dreamboyn81** [comment]
![41ae7901213fb80ecd0c15cd3ed12f2eb8389496](https://cloud.githubusercontent.com/assets/15012771/16543934/4ab3d260-4123-11e6-8785-71a83a99b74e.jpg)
![274e893d70cf3bc767e0b9f6d900baa1cc112a8b](https://cloud.githubusercontent.com/assets/15012771/16543938/65eef654-4123-11e6-8b70-e4652c900ca1.jpg)

**Comment by nfsvsgta** [comment]
What's the game?

**Comment by dreamboyn81** [comment]
Fire Emblem：Path Of Radiance

**Comment by mk8itra1n** [comment]
Sonic Adventure 2 and Sonic Adventure Only make it to their title screens but I'm unable to go further than that! Sonic Mega Collection actually gets ingame but the games play in extreme slow motion

**Comment by nfsvsgta** [comment]
Anyway EID Mubarak

**Comment by GingerOfOz** [comment]
Here's an essay for you guys, sorry it's so long, but I figured any info might be useful from my testing.  I am on a Jailbroken iOS 9.0.1 iPad Air 2, and I decided to put the rest of my 64 gigs of space to good use.  I transferred my games to my iPad, and ran some tests on all of them.  Here were my results:

GENERAL ISSUES FOR ALL GAMES:
- Onscreen controller worked for the most part, but the main control stick did nothing, and the d-pad reacted like I was pushing A.  This restricted much of the testing I could do, because I could not format the Memory Card.
- Some games didn't think there was a memory card to use.

Here is results from individuals games.

4 Swords Adventure: Runs fine until choice between shadow battle and campaign.  Onscreen controller then does nothing. 40 fps start screen.

Wind Waker: Nintendo logo shows up, fades, and then black screen.

Luigi's Mansion: fine until memory card issue.  60 fps load screen

Mario Kart Double Dash!: White screen with controller for 15 seconds, then black screen with controller.  Fps jumps around a bit but GC4iOS crashes after a few seconds.

Mario Party 5: 30-60 FPS throughout entire load/ start screen.  Has memory card issue, but I was able to start a party successfully.  Graphical glitch around any sort of text box, but I get that in Dolphin 5.0, so no worries.  

Mario Party 6: Crashes/ Freezes in random places fairly often.  Same text box graphics glitch. I am able to start a party.  

Mario Party 7: Fairly slow, generally about 10 FPS. I am able to start a party. (Picture of glitch included)

Pikmin: Cant play due to d-pad and control stick not working, so I can't format mem card.

Pikmin 2: Can't get past the Nintendo logo without crashing.

Super Mario Sunshine: 40-60 FPS through loading and short movie before start screen!  Unfortunately, after pressing a button, or watching to where the start screen should show up gives nothing but a black screen.

Smash Bros. Melee: Can't test because half of the time I get "An error has occurred.  Turn off your Nintendo GameCube and check the user manual."  The rest of the time, I get the "Format the Memory Card in slot A", but after a couple seconds it says "There is no memory card in slot A, and won't move from that screen. ( picture included)

Hope this helps someone, and I do understand that this is all Aplha.
![image](https://cloud.githubusercontent.com/assets/20305474/16596475/fb4b2b6a-42a8-11e6-9595-401f50a5c3dd.png)
![image](https://cloud.githubusercontent.com/assets/20305474/16596477/fb66404e-42a8-11e6-8eef-c8f7de4699c6.png)
![image](https://cloud.githubusercontent.com/assets/20305474/16596476/fb65b246-42a8-11e6-8b51-7623153c2042.png)
![image](https://cloud.githubusercontent.com/assets/20305474/16596478/fb67a09c-42a8-11e6-9ac8-3fa4a9be2ae6.png)

**Comment by GingerOfOz** [comment]
*Second picture is from Pikmin. Pushing retry led me back to the start screen.

**Comment by Fadi5555** [comment]
@GingerOfOz wow thank you so much for this essay. I like your testing of games.
Please can you test this game for me please? (Crash bandicoot the warth of cortex, resident evil 4, crash nitro kart, spyro the new beginning) please test the and tell us the results with pictures please.
Also if you have an YouTube channel to see your tests it will be amazing.
Thanks a lot.

**Comment by zakusa** [comment]
ok ginger from now your williams tester i would test but i dont have jailbroken ios or mac os x to get xcode i even tried vmware it wouldnt let me connect ipad to it but i do have a laptop which i can dual boot mac os x maybe but i dont know how to dualboot mac os x

**Comment by GingerOfOz** [comment]
@Fadi5555 I own the Spyro game, so I'll run that one for you.  I'm looking into different screen recorders right now.

**Comment by Fadi5555** [comment]
Oh thank you so much for your politeness. Also if you can don't forget to test those games that I mentioned previously please. I also own ipad air 2 but it on 9.3.2 and waiting for upcoming jailbreak. a i hope next jaibreak will defeat KPP (Kernel Patch Protection) if this happen we will get JIT support.
This is important to run emulators in fullspeed like (ppsspp, retroarch, and of course GC4ios as well.

**Comment by GingerOfOz** [comment]
Memory card issue is fixed.  Controller is more fixed, but the control stick often gets stuck going one direction, and won't go any direction but that.

Still an improvement though.

**Comment by GingerOfOz** [comment]
Did some more testing.  Here's what's new.

GENERAL ISSUES:
- I noticed that a couple of people mentioned they had sound.  I don't care about sound all that much, but if I'm supposed to be getting it I'm not.
- D-Pad (The plus button) still only reacts as if I were pressing A.
- Like I mentioned earlier, the main controller sticks, and it always sticks to the right.  I am still able to go up and down, but going right is impossible.  This happens in every game the moment I move the control stick.

4 Swords Adventure: I find that I can actually still use the controller, if I move before the link picture fades into existence on the right side of the screen.  I can go into options and change them, but clicking on "Adventure" or "Shadow Battle" causes the game to freeze.

No change on The Wind Waker.

Luigi's Mansion:  I can now create save files without issues.  Clicking on a save file leads me to the short startup movie, BUT pushing a button to exit it or waiting until it ends causes it to hit a black screen.  This is EXACTLY like the issue that Super Mario Sunshine has, which makes me wonder if the emulator has an issue with transferring from a movie into the next part, especially since both games are made straight from Nintendo.

No change on Double Dash

Mario Party 5/6/7: I can now create and use save files.  

Pikmin 1: Appears to now be playable.  I can create a save file, and it loads up the movie just fine.  I'm not sure if it will crash after the movie like Luigi's Mansion and Super Mario Sunshine, so I will try that when I have time to sit through a 2 FPS movie of a rocket flying through space.

Pikmin 2: No change, except that sometimes it reaches the point where it says Nintendo 2004 at the bottom, and the leaves SHOULD start fading into view, but it just freezes.

No change on Super Mario Sunshine

Super Smash Bros. Melee: I now have a new issue.  I still get the "Error has occurred.  Power off GameCube and check the user manual" most of the time.  The rest of the time I get the other message pictured, and then nothing changes.

That's all I've got for now.  I'll do some research on the Smash Bros. issue, see how Pikmin does, and get back to you guys.

Thanks for your work!
![image](https://cloud.githubusercontent.com/assets/20305474/16640457/68028cdc-43b5-11e6-93dc-0460c40da7d8.png)
![image](https://cloud.githubusercontent.com/assets/20305474/16640458/681e72f8-43b5-11e6-990f-8ef08c5667c0.png)

**Comment by Fadi5555** [comment]
Please make a test for crash bandicoot the warth of cortex and resident evil 4 please?
This is my request for you. I'll waiting for your results as well as screenshots please.

**Comment by nfsvsgta** [comment]
Just a reminder of Need For Speeds not working ie showing black screen only

**Comment by Fadi5555** [comment]
I hope the creator of this emu could fix black screen of some games including need for speed.

**Comment by nfsvsgta** [comment]
Need for speed <3

**Comment by nfsvsgta** [comment]
Unlike many m a big fan of nfs

**Comment by jakeb072001** [comment]
I haven't sent anything for a while now and that's because my internet has been doing some weird stuff...

But I can finally send some info now!

So one thing is that whenever you get a black screen just wait a while (usually 5 minute or more) and you will eventually see the game!
In Luigis mansion when it hits black screen just wait and eventually you will have the game but you can't play yet for multiple reasons, It's extremely slow and It has the joystick issue.

Sound seems to work fine and saving seems to work fine (at least for me).

**Comment by nfsvsgta** [comment]
I've waited for 10 mins for nfs probably more would do. I'll give it a try soon

**Comment by GingerOfOz** [comment]
I've tried the wait on Super Mario Sunshine, by I'll give it another shot on some of the other ones.

Also, I'm only going to test games that I own.  I get that I COULD go and download them, but I think the SHOULD part is also important.

**Comment by DarknesGaming** [comment]
Wow. You people must really enjoy GC4iOS, and constantly running into problems with it. IS this really going to be Dolphin for iOS? I'm surprised Nintendo has not tried to shut us down yet.

**Comment by zakusa** [comment]
ok guys i know this is in beta but is it enough to play smash bros melee full speed on ipad air 2

**Comment by jakeb072001** [comment]
I have found a few things that where (kind of) hidden:
1. /var/mobile/Library/Load (texture packs) (I have put a Luigis mansion texture pack in it but I'm not sure if it works yet or not (I can't tell sd from hd...))
2. /var/mobile/Library/StateSaves (save states)
3. /var/mobile/Library/Wii (wii stuff)
4. /var/mobile/Library/GC (memory card saves and stuff) (this is where you can put your saves from pc so that you can play your games on the go without losing progress (this is very useful for games like animal crossing))

I say these where kind of hidden because they weren't invisible but they where still scattered and in a weird location that jailed users could not get to

Edit: also should we create a new issue titled "General Discussion" and post all generic stuff in there? Just an idea.

**Comment by newhacker1746** [comment]
hey guys gone for a while but im back. Downloading latest

**Comment by newhacker1746** [comment]
Starfield working. Had to delete mario due to space issues. Will test when I get back home. Thanks guys for all the work put into it, its doing great.

**Comment by DarknesGaming** [comment]
I can't seem to import any of my ISOs using iTunes or iFunbox. It's not too late to add File Sharing capabilities.

**Comment by DarknesGaming** [comment]
I just realized I don't have a lot of experience in the Xcode field, but I figured out how to add a scene, and program a button to to go there.
I can't program the button there to go back to the main screen. We are doomed.
Edit: It's too bad I can't learn how to use Xcode from William L. Cobb himself. I really want to contribute to GC4iOS and know what I'm actually doing.

**Comment by GingerOfOz** [comment]
I tried waiting out the black screen again, and this time I got results.  It eventually loaded the "Press Start" screen on Mario Sunshine, and Luigi's Mansion also got past the black screen. Wind Waker took considerably longer, but I got to where you could see the sun start to show up. After a few seconds though, the emulator crashed on all of these games.

When I get home to my computer, I'll try putting my memory card in the emulator and report back.

**Comment by DarknesGaming** [comment]
Nice job @GingerOfOz for using patience. People these days aren't patient enough to get results, making you NOT one of them.
I'm not sure using the OpenGL backend is a good idea. It crashed on my computer running a dedicated GPU. How is GC4iOS going to run on iOS if OpenGL is the only backend available?
Edit: There aren't any settings for GC4iOS. I'm trying to figure out how to add that, and link the settings the same way @WilliamLCobb did.

**Comment by GingerOfOz** [comment]
@jakeb072001  A quick change to your last post, you have to go to /var/USER/Library/... In order to get to the save states/ textures/ mem cards.  I put mine in from my laptop and I'll see how it takes it.

**Comment by GingerOfOz** [comment]
Memory card worked!  It loaded up my data just fine.

**Comment by DarknesGaming** [comment]
How are you able to do that? Are you using a jailbroken device?
I'm trying to enable iTunes File Sharing for GC4iOS. If the app says you can import files with iTunes, that file sharing ability needs to be enabled.
Edit: Let's see if I can give GC4iOS an ability to unzip 7z files.

**Comment by GingerOfOz** [comment]
Yeah, I'm Jailbroken.

Transferring files through iTunes didn't work for me either, and Dropbox/ Google Drive took WAY too long.  I use an application called iExplorer on my laptop to transfer files super quickly and painlessly.  I don't know if you can access its full capabilities without Jailbreak though.  The free version has everything that I need.

**Comment by DarknesGaming** [comment]
I use iFunbox, but I think I'll try out iExplorer. Plus, I think Dropbox and Google Drive are better when you're syncing the memory cards.

**Comment by DarknesGaming** [comment]
I have successfully managed to build GC4iOS version 0.1.3: iTunes File Sharing capabilities. If this doesn't make @WilliamLCobb proud, then I will immediately start adding the 7z unzipping ability.
Edit: Unfortunately, I had to completely remove armv7 support. Any device that is 32-bit will NOT be able to run GC4iOS. This includes the iPhone 5C, iPad 4, iPad mini, iPod Touch 5, and EVERY device older than those. If I kept armv7 support, I would keep running into errors.
Edit 2: For those who want to testify against my actions, read the README.md file. The file clearly says it only supports 64-bit devices, which means the devices stated in my last edit can't run this. Although this may come at a disadvantage for those with older phones, this makes it less likely to run into any errors whatsoever.
Edit 3: As of Dolphin 5.0, 32-bit devices aren't supported, and @WilliamLCobb most likely has an intention to bring the features to iOS. The older versions of Dolphin were too slow.
Edit 4: I just tried to run a game on a device with only 1 GB of RAM. The app crashed instantly. Maybe there's a way to make the app allocate less than 1 GB?

**Comment by nfsvsgta** [comment]
Half an hour gone and still black screen on nfs underground.

**Comment by DarknesGaming** [comment]
You seem to be new, @nfsvsgta. Unfortunately, GC4iOS is an unstable version of mobile Dolphin ported to iOS, and if I were any smarter, I would say that your Apple device is preventing JIT from running. We might have to switch to the Interpreter for this.
I talked to a programmer, and he told me that if you want JIT to run, you need to find an exploit that will allow JIT to run in the memory heap, where all of the root processes are allocated.

**Comment by mk8itra1n** [comment]
@WilliamLCobb it's been a while brother. Any update on GC4iOS?

**Comment by newhacker1746** [comment]
Yeah it has been. Mario runs now at 10fps max.

**Comment by DarknesGaming** [comment]
The frame skip has been set to 5 on GC4iOS, so the app will obviously run at 10 fps. Honestly, I don't think mobile devices and tablets can handle consoles like the GameCube.

**Comment by newhacker1746** [comment]
You're right, I also uss the PAL edition (It's cheaper than ntsc on ebay in U.S), And supports PAL60/EURGB basically 60fps for europe, and works easy with homebrew wii. Regular dolphin can handle pal60, not gc4ios but im looking through documents/dolphin to see if I can find a setting.

**Comment by mk8itra1n** [comment]
@WillamLCobb it's been awhile brother. Hope your well and everything. Just interested to know about if you got any updates about GC4iOS? Thanks

**Comment by newhacker1746** [comment]
I've been fidgeting with my fingers about an update too

**Comment by Fadi5555** [comment]
Me too we all waiting for new update to see what's new.
It's really have been long time we didn't see anything from @WilliamLCobb. That's weird?

**Comment by GingerOfOz** [comment]
I think he's been spending most of his time on Pokemon GO hacks.  Probably about a third of my repos have it, and they keep having new features added.  So he might not come back to this for a little while.

**Comment by Fadi5555** [comment]
That's too sad.
Because if he works in this project regularly we will matching other platforms of dolphin port.
Even we could bringing all features of dolphin 5.0 to iOS port.
I really hope he come back and continue this amazing project.

**Comment by WilliamLCobb** [comment]
Just to update: Currently the iOS version has every feature working except for a few advanced significant speed ups (fastmem and a few on the GPU side). JIT and OpenGL are work 100%. The reason it's still so slow is poor GPU drivers and a CPU scheduling problem (CPU is only running at like 20% capacity).

I really don't know enough to fix the CPU scheduling problem. I was hoping someone more familiar with Dolphin could help but no updates so far.

If any of these things are fixed I'll update the repo right away. If no one else works on it I'll try fixing it when I get some more free time.

**Comment by WilliamLCobb** [comment]
And iOS was integrated in a way that any new features should automatically work. If not it will be easy to add them

**Comment by zakusa** [comment]
That sucks I thought the emulator was almost done

**Comment by DarknesGaming** [comment]
I continue to sit and wait for GC4iOS to be just as powerful as Dolphin 5.0.
The lowest resolution Dolphin supports would fit on all 64-bit iOS devices. Speaking of 64-bit, still waiting on @WilliamLCobb to merge my work.

**Comment by DarknesGaming** [comment]
Also, Dolphin requires a dedicated GPU if you want adequate performance. Apple integrates their own GPU, and since the GPU is only powerful enough for mobile apps, that would not be able to run a desktop emulator.

**Comment by newhacker1746** [comment]
Interesting. But I think apple started making better gpu's since the iPhone 6 and Apple TV 4?

**Comment by TheTesterMC** [comment]
i tried To Open Kirby Air Ride because i thinked it was able to run it, because the game in my laptop(that is a crappy laptop) the game opened but it get stucked whit a black screen when i open it
(sorry if my vocabulary is bad, but is because i speak spanish)

**Comment by zakusa** [comment]
ya i know it sucks no one wants to help william cobb

**Comment by newhacker1746** [comment]
I'd love to, but unfortunately my knowledge of coding and emulation is almost none. I know best xml, but even then I couldn't fix much with that knowledge, if anything. Just hope that William gets some help or completes the merge into Dolphin to get the main team's help.

**Comment by starbladex** [comment]
@WilliamLCobb please support on apple tv 4

**Comment by zakusa** [comment]
So ya did you guys see the cool ps2 emulator that's in beta right now

**Comment by zakusa** [comment]
https://www.youtube.com/watch?v=RixW3pCENe4

**Comment by nfsvsgta** [comment]
![image](https://cloud.githubusercontent.com/assets/19260668/17558139/a50cdfa4-5ee7-11e6-9d21-e08a19380a54.jpg)

**Comment by nfsvsgta** [comment]
Nfs underground on Ps2 emulator on iOS

**Comment by zakusa** [comment]
ya ps2 sucks most of the games can be downloaded on app store, psp, and wii instead he should help will cobb with this emulator and get wii working cause wii has more games and better games than ps2

**Comment by zakusa** [comment]
@WilliamLCobb ya it's been 26 days I want to know how things went

**Comment by DarknesGaming** [comment]
Maybe @WilliamLCobb is busy working on his Poke Go++ hack. I could carry on the work of GC4iOS, but that would take me a LOT of time.

**Comment by zakusa** [comment]
Ya I don't like Pokemon go I would learn how to create emulator but i want to learn c++ instead of Xcode cause Apple hardware sucks and ya there's alotttt to study at school I come home and got no time to learn c++

**Comment by brand175** [comment]
if a ios 10 jailbreak came out would gc4ios still work?

**Comment by zakusa** [comment]
brand gc4ios is deadddddd even if it works on iOS 10 it's useless it can't play any games full speed

**Comment by brand175** [comment]
![img_0008](https://cloud.githubusercontent.com/assets/20964160/18541190/63e64436-7ad8-11e6-9b1a-d850839d7154.PNG)

I have gameplay on disneys pk out of the shadow. Playing on level 2,3,4,5,6 with no crash but the controls are ass. https://youtu.be/sfNFZ-x28O0    I have lots and lots of testing videos and I have some gameplay videos. When gc4ios is more stable I'm planing on doing 2 hour videos of each GameCube and wii game I own. Even if it's 0.005 fps... I'm patience and willing to test. Youtube=brand175.  
Here's my gc4ios gameplay playlist http://www.youtube.com/playlist?list=PL6i5t6QRkh2uUdp0-ymvjQ9mABiFzFOlZ
Here's my gc4ios testing playlist http://www.youtube.com/playlist?list=PL6i5t6QRkh2t_1-J5bJYZqH03g_o3E2wW
Here's my gc4ios wii testing playlist http://www.youtube.com/playlist?list=PL6i5t6QRkh2u1pE4hNDAjGPQ-WFzjgBYi

**Comment by brand175** [comment]
![img_0009](https://cloud.githubusercontent.com/assets/20964160/18541298/0cb5fc96-7ad9-11e6-90b2-5c101d213825.PNG)
Wii sports gets to the title then buttons don't work
![img_0010](https://cloud.githubusercontent.com/assets/20964160/18541355/7f6a142a-7ad9-11e6-96b1-b03d246d1111.PNG)
Mario party 8 does the same thing
![img_0011](https://cloud.githubusercontent.com/assets/20964160/18541648/4802ad7e-7adb-11e6-841d-0424fd3b7da8.PNG)
pain in the ass to boot but it works ok no gameplay though :(
![img_0012](https://cloud.githubusercontent.com/assets/20964160/18541676/8c8a0050-7adb-11e6-9dff-338ed9e23bc2.PNG)
AAAHH you can only view the collab of past info of the metroid games :(
![img_0013](https://cloud.githubusercontent.com/assets/20964160/18541720/df55a7da-7adb-11e6-84f6-d4dee3e2c2b8.PNG)
Menus work but no gameplay :(
![img_0014](https://cloud.githubusercontent.com/assets/20964160/18541833/790b3e58-7adc-11e6-9508-be5cab1aca93.PNG)
More pain in the ass than Medal of honour to boot but it works just don't reload at the toaster part 😭
![img_0015](https://cloud.githubusercontent.com/assets/20964160/18542091/cba0648a-7add-11e6-8f82-a8420ebaabdc.PNG)
This took like 10 minutes to pass that stupid mem card with the floating golf ball but if you have to format then It's another 8 minutes
![img_0016](https://cloud.githubusercontent.com/assets/20964160/18542773/6d831100-7ae1-11e6-81b0-9db76ad123e2.PNG)
Metroid prime 2 echoes boots but can't go in the intro 😭The first metroid prime won't boot at all 😥
![img_0017](https://cloud.githubusercontent.com/assets/20964160/18542873/fe501926-7ae1-11e6-85f3-8bf4a28ec9b4.PNG)
Only the intro works 😡

**Comment by Fadi5555** [comment]
It's been long time William didn't appear and continue his work on this amazing project. I hope to continue improving it

**Comment by zakusa** [comment]
ya its saddddd i thought this was almost done

**Comment by zakusa** [comment]
all i can play is smash bros n64 on my ipad and it sucks i want smash bros gamecube on ipad its sooooo much better

**Comment by brand175** [comment]
https://www.dropbox.com/s/pc57vmza8wmud95/GC4iOS%20%28Cached%20Interpreter%29.ipa?dl=0
